lib/auth/v2/getCanonicalizedResource.js

```diff
@@ -24,7 +24,23 @@ function getCanonicalizedResource(request) {
     */
 
     // Specified subresources:
-    const subresources = [
+    const subresources = clientType === 'GCP' ?
+    [
```

Comment:
you can optionally store the resources for GCP and for aws in separate variables, and create an object map:
```
const subresourcesByType = {
    GCP: gcpSubresources,
    AWS: awsSubresources
};
const subresources = subresourcesByType[clientType];
```
which may be a little easier to read and extend if more client types are added down the line. But it's a minor suggestion, not anything important.

Reply:
will take note of that and make the modification for readability.